pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -479,6 +486,39 @@ func (p *ConfigurationEditFlags) Apply(
 		service.Spec.Template.Spec.TimeoutSeconds = &p.TimeoutSeconds
 	}
 
+	if cmd.Flags().Changed("profile") {
+		profileName := ""
+		deleteProfile := false
+		if strings.HasSuffix(p.Profile, "-") {
+			profileName = p.Profile[:len(p.Profile)-1]
+			deleteProfile = true
```

Comment:
It would be nice to have this case covered in unit test.

Reply:
I have added unit tests.